base/broadcast.jl

```diff
@@ -471,6 +485,14 @@ end
 .+(As::AbstractArray...) = broadcast_elwise_op(+, As...)
 .*(As::AbstractArray...) = broadcast_elwise_op(*, As...)
 
+# fallbacks for non-AbstractArray
```

Comment:
IIRC @stevengj already plans to move all of these to `broadcast` too. Steven, is this OK for now?